discussment-ui-wicket/src/main/java/org/danekja/discussment/ui/wicket/panel/postReputation/PostReputationPanel.java

```diff
@@ -62,21 +66,21 @@ protected void onInitialize() {
 
         add(new Label("dislikes", new PropertyModel(postReputationModel, "dislikes")));
 
-        Form prform = new PostReputationForm("prform", postModel, userService, postReputationService, accessControlService);
+        add(new PostReputationForm("prform", postModel, postService, postReputationService, userService, accessControlService));
 
         Label liked = new Label("liked");
-        liked.setVisible(false);
+        add(liked);
 
-        if (postReputationService.userVotedOn(userService.getCurrentlyLoggedUser(), postModel.getObject())){
-            if (postReputationService.userLiked(userService.getCurrentlyLoggedUser(), postModel.getObject())){
-                liked = new Label("liked", new ResourceModel("postReputation.liked"));
+        if (!userService.isGuest() && postReputationService.userVotedOn(userService.getCurrentlyLoggedUser(), postModel.getObject())) {
```

Comment:
I believe this should be replaced by a Model which will be passed to the Label(liked) (and inside will check what to show) based on postReputationService.getVote result.

Visibility of liked Label should be decided by onConfigure method override which will check isGuest and userVotedOn to decide whether to set visible or not.

Current implementation wont work properly when e.g. user´s session expires.

Reply:
True.